File: backend/Models/Domain/Theme.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TestScriptTracker.Models.Domain
{
    [Table("Themes")]
    public class Theme
    {
        [Key]
        public Guid ThemeId { get; set; }
        [MaxLength(30)]
        public string ThemeName { get; set; } = string.Empty;
        public int FontSize { get; set; }
        public bool IsDeleted { get; set; } = false;
        public Guid ClientId { get; set; }//FK
        public int FontId { get; set; }//FK

        public Client? Client { get; set; }//(m:Theme, 1:Client)
        public Font? Font { get; set; }//(m:Themes, 1:Font)
        public List<Logo>? Logos { get; set; }//(1:Theme, m:Logos)
        public List<ColourScheme>? ColourSchemes { get; set; }//(1:Theme, m:ColourScheme)


    }
}

File: frontend/src/app/models/theme.ts
export interface Theme {
    themeId?: string;
    themeName: string;
    fontSize: number;
    clientId: string;
    fontId: number;
    logos: Logo[];
    colourSchemes: ColourScheme[];
  }
  
  export interface Logo {
    logoId?: string;
    logoImage: string;
    themeId: string;
  }
  
  export interface ColourScheme {
    colourSchemeId?: string;
    colour: string;
    themeId: string;
  }
  
  export interface Font {
    fontId: number;
    fontName: string;
  }

  export interface ThemeDetails {
    theme: Theme;
    themeLogos: Logo[];
    themeColourSchemes: ColourScheme[];
  }
  
  export interface ThemeDto {
    themeId: string;
    themeName: string;
    fontSize: number;
    clientId: string;
    fontId: number;
  }

  //excludes the themeId as it is sent via the route/header of the request
  export interface UpdateThemeDto {
    themeName: string;
    fontSize: number;
    clientId: string;
    fontId: number;
  }